### DISCUSSION TITLE
Using INIConfig

### DISCUSSION THREAD

**Discussion by hbarnard** [mojolicious/mojo#2125] [Q&A] [answered]
I've decided, perhaps unwisely, to convert from a stored hash image and the standard config plugin to INIConfig: https://metacpan.org/pod/Mojolicious::Plugin::INIConfig

So I have:
```
 if ( -e $config_file ) {
		$config  = $self->plugin('INIConfig' => 
		              {file => $config_file},
		              ) ;
```
which puts the config nicely into $self:
```
 'defaults' => {
                                 'config' => {
                                               '_' => {
                                                        'mailpath' => '/var/spool/mail',
```
but I need this:

`$self->app->config->{'mailpath'}`

because the rest of the application (right or wrong) is coded that way. So how do I load `$self->app->config` instead?

Thanks in advance Hugh B

**Comment by Grinnz** [comment, marked as answer]
It is also loaded into `app->config`, have you tried it? It will parse it into the subsection, in your example underscore which is the subsection Config::Tiny parses the main section into: `app->config->{_}{mailpath}`

**Reply by hbarnard** [reply]
Yes, thanks for answering so quickly. There it is. Some recoding but the conf files will be more manageable for non-specialists. It's this: https://sourceforge.net/projects/cclite2/ which is gradually inching forwards. Thanks again.
